internal/conf/jwk.go

```diff
@@ -0,0 +1,154 @@
+package conf
+
+import (
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+)
+
+type JwtKeysDecoder map[string]JwkInfo
+
+type JwkInfo struct {
+	PublicKey  jwk.Key `json:"public_key"`
+	PrivateKey jwk.Key `json:"private_key"`
+}
+
+// KeyInfo is used to store the initial config of the keys
+// The private key should be in DER format and base64 encoded
+type KeyInfo struct {
+	Type       string `json:"type"`
+	PrivateKey string `json:"private_key"`
+	InUse      bool   `json:"in_use"`
+}
+
+// Decode implements the Decoder interface
+// which transforms the keys stored as der binary strings into jwks
+func (j *JwtKeysDecoder) Decode(value string) error {
+	data := map[string]KeyInfo{}
+	if err := json.Unmarshal([]byte(value), &data); err != nil {
+		return err
+	}
+
+	config := JwtKeysDecoder{}
+	for kid, key := range data {
+		// all private keys should be stored as der binary strings in base64
+		derBytes, err := base64.StdEncoding.DecodeString(key.PrivateKey)
+		if err != nil {
+			return err
+		}
+
+		var privKey any
+		if key.Type == "hmac" {
+			privKey = derBytes
```

Comment:
haven't found a better way to determine if the key is a hmac key just based off the DER string, so i've decided to add a `type` field to indicate this

Reply:
HMAC just uses octal keys, so `kty` will be `oct` with a `k` property, like so:

```json
{
  "kty": "oct",
  "k": "<Base64-URL of the key bytes>",
  "use": "sig"
}
```

That should be enough to identify it.